project_invoicing/models/project_project.py

```diff
@@ -19,12 +19,22 @@ class ProjectProject(models.Model):
     def generate_invoices(self, data):
         self.ensure_one()
 
-        lines_by_partner = defaultdict(list)
+        self = self.with_context(
+            force_company=self.company_id.id,
```

Comment:
to ensure that the correct values are taken for multi-company properties used (on partners and products) when generating the invoice. This ensures that a user with multi company access may trigger an invoice from a project related to another company.